consensus/src/quorum_store/proof_manager.rs

```diff
@@ -232,26 +117,28 @@ impl ProofManager {
                     request.block_timestamp,
                 );
 
-                counters::NUM_BATCHES_WITHOUT_PROOF_OF_STORE.observe(self.batch_queue.len() as f64);
+                counters::NUM_BATCHES_WITHOUT_PROOF_OF_STORE
+                    .observe(self.batch_proof_queue.num_batches_without_proof() as f64);
                 counters::PROOF_QUEUE_FULLY_UTILIZED
                     .observe(if proof_queue_fully_utilized { 1.0 } else { 0.0 });
 
-                let excluded_batches: Vec<_> = excluded_batches.iter().cloned().collect();
-
                 let (opt_batches, opt_batch_txns_size) = if self.enable_opt_quorum_store {
                     // TODO(ibalajiarun): Support unique txn calculation
                     let max_opt_batch_txns_size = request.max_txns - txns_with_proof_size;
-                    let (opt_batches, size) = self
-                        .batch_queue
-                        .pull_batches(max_opt_batch_txns_size, excluded_batches.clone());
-
-                    (
-                        opt_batches
-                            .into_iter()
-                            .map(|(batch_info, _)| batch_info)
+                    let (opt_batches, opt_payload_size, _) = self.batch_proof_queue.pull_batches(
+                        &excluded_batches
+                            .iter()
+                            .cloned()
+                            .chain(proof_block.iter().map(|proof| proof.info().clone()))
                             .collect(),
-                        size,
-                    )
+                        max_opt_batch_txns_size,
+                        request.max_txns_after_filtering,
+                        request.soft_max_txns_after_filtering,
```

Comment:
At this point, the block proposal already contains a bunch of proofs. 
We need to subtract these transactions from `request.max_txns_after_filtering` and ` request.soft_max_txns_after_filtering`.

Reply:
There is a TODO for that in L126. We still need to flush out some details, so it's left like that.

Reply:
Cool. Let's fix that before landing? Otherwise, for some forge tests, the block size might be too big, and the test may fail.

Reply:
I don't think it will affect any existing tests. It's behind a feature flag and disabled by default. `enable_opt_quorum_store`